libweasyl/libweasyl/models/media.py

```diff
@@ -92,47 +98,30 @@ def make_thumbnail(self, source_image=None):
         if thumbnail is source_image:
             return self
         else:
-            return fetch_or_create_media_item(
+            return self.fetch_or_create(
                 thumbnail.to_buffer(format=self.file_type.encode()), file_type=self.file_type,
                 im=thumbnail)
 
-
-class DiskMediaItem(MediaItem):
-    __table__ = tables.disk_media
-    __mapper_args__ = dict(polymorphic_identity='disk')
-
-    def init_from_data(self, data):
-        path = ['static', 'media'] + fanout(self.sha256, (2, 2, 2)) + ['%s.%s' % (self.sha256, self.file_type)]
-        self.file_path = os.path.join(*path)
-        self.file_url = '/' + self.file_path
-        real_path = self.full_file_path
-        makedirs_exist_ok(os.path.dirname(real_path))
-        with open(real_path, 'wb') as outfile:
-            outfile.write(data)
-
     @property
     def display_url(self):
         # Dodge a silly AdBlock rule
         return self.file_url.replace('media/ad', 'media/ax')
 
-    # set by configure_libweasyl
-    _base_file_path = None
-
     @property
     def full_file_path(self):
         return os.path.join(self._base_file_path, self.file_path)
 
-    def serialize(self, recursive=1, link=None):
-        ret = super(DiskMediaItem, self).serialize(recursive=recursive, link=link)
-        ret['full_file_path'] = self.full_file_path
-        return ret
-
     def as_image(self):
         return images.read(self.full_file_path.encode())
 
+    @property
+    def file_path(self):
+        path = ['static', 'media'] + fanout(self.sha256, (2, 2, 2)) + ['%s.%s' % (self.sha256, self.file_type)]
+        return os.path.join(*path)
 
-def fetch_or_create_media_item(*a, **kw):
-    return DiskMediaItem.fetch_or_create(*a, **kw)
+    @property
+    def file_url(self):
+        return '/' + self.file_path
```

Comment:
For Windows compatibility and a general sense of correctness, there should probably be a third property involved that returns a list of path components. `file_path` can do `os.path.join` and `file_url` can do `'/' + '/'.join`. (Or `full_file_path` could even contain all of `file_path`, since it isn’t used anywhere else.)